rewrite-java/src/main/java/org/openrewrite/java/RenamePackage.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import lombok.*;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Option;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.nio.file.Paths;
+
+import static org.openrewrite.Tree.randomId;
+
+/**
+ * A recipe that will rename a package name in package statements, imports, and fully-qualified types (see: NOTE).
+ *
+ * Examples:
+ *  - org.openrewrite -> org.openrewrite.java
+ *  - org.openrewrite.java -> org.openrewrite
+ *  - javax.xml.bind -> jakarta.xml.bind
+ *
+ * NOTE: Does not currently transform all possible type references, and accomplishing this would be non-trivial.
+ * For example, a method invocation select might refer to field `A a` whose type has now changed to `A2`, and so the type
+ * on the select should change as well. But how do we identify the set of all method selects which refer to `a`? Suppose
+ * it were prefixed like `this.a`, or `MyClass.this.a`, or indirectly via a separate method call like `getA()` where `getA()`
+ * is defined on the super class.
```

Comment:
@jkschneider most of the code is from your ChangeType and ChangePackage recipes. I'm not sure how to reference you're essentially the original author. Do you have any recommendations?